cpp/tests/test_abm_infection.cpp

```diff
@@ -187,79 +185,75 @@ TEST(TestInfection, getPersonalProtectiveFactor)
         mio::abm::ExposureType::GenericVaccine, mio::AgeGroup(0), mio::abm::VirusVariant::Wildtype}](0);
     auto defaut_severity_protection  = params.get<mio::abm::SeverityProtectionFactor>()[{
         mio::abm::ExposureType::GenericVaccine, mio::AgeGroup(0), mio::abm::VirusVariant::Wildtype}](0);
-    ASSERT_NEAR(defaut_infection_protection, 0, 0.0001);
-    ASSERT_NEAR(defaut_severity_protection, 0, 0.0001);
+    EXPECT_NEAR(defaut_infection_protection, 0, eps);
+    EXPECT_NEAR(defaut_severity_protection, 0, eps);
 
     // Test linear interpolation with one node
-    mio::set_log_level(mio::LogLevel::critical); //this throws an error either way
     params.get<mio::abm::InfectionProtectionFactor>()[{mio::abm::ExposureType::GenericVaccine, person.get_age(),
                                                        mio::abm::VirusVariant::Wildtype}] =
-        [](ScalarType days) -> ScalarType {
-        return mio::linear_interpolation_of_data_set<ScalarType, ScalarType>({{2, 0.91}}, days);
-    };
+        mio::TimeSeriesFunctor<ScalarType>{mio::TimeSeriesFunctorType::LinearInterpolation, {{2, 0.91}}};
     auto t = mio::abm::TimePoint(6 * 24 * 60 * 60);
-    ASSERT_NEAR(person.get_protection_factor(t, mio::abm::VirusVariant::Wildtype, params), 0, 0.001);
-    mio::set_log_level(mio::LogLevel::warn); //this throws an error either way
+    // TODO: Discuss: Assumption of interpolation in TDPF is that the function is constant with value at front/back entry outside of [front, back] time range. This works with one node as well and prints no errors
```

Comment:
I think having it constant before and after makes sense for linear interpolation. The dropoff to 0 doesn't make much sense. If the user wants to have constant 0, this can be implemented by the user by an additional time point in the linear interpolation (see other TODOs).